cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/authentication/handler/support/ClientAuthenticationHandler.java

```diff
@@ -18,67 +18,80 @@
  * @since 3.5.0
  */
-public class ClientAuthenticationHandler extends AbstractClientAuthenticationHandler {
+@Component("clientAuthenticationHandler")
+@SuppressWarnings("unchecked")
+public class ClientAuthenticationHandler extends AbstractPac4jAuthenticationHandler {
 
     /**
-     * Whether to use the typed identifier (by default) or just the identifier.
+     * The clients for authentication.
      */
-    private boolean typedIdUsed = true;
+    @NotNull
+    @Autowired
+    private Clients clients;
 
-    /**
-     * Define the clients.
-     *
-     * @param theClients The clients for authentication
-     */
-    public ClientAuthenticationHandler(final Clients theClients) {
-        super(theClients);
+    @Override
+    public final boolean supports(final Credential credential) {
+        return credential != null && ClientCredential.class.isAssignableFrom(credential.getClass());
     }
 
-    /**
-     * {@inheritDoc}
-     */
     @Override
-    protected HandlerResult createResult(final ClientCredential credentials, final UserProfile profile)
-        throws GeneralSecurityException, PreventedException {
-        final String id;
-        if (typedIdUsed) {
-            id = profile.getTypedId();
-        } else {
-            id = profile.getId();
-        }
-        if (StringUtils.isNotBlank(id)) {
-            credentials.setUserProfile(profile);
-            return new DefaultHandlerResult(
-                this,
-                new BasicCredentialMetaData(credentials),
-                this.principalFactory.createPrincipal(id, profile.getAttributes()));
-        }
-        throw new FailedLoginException("No identifier found for this user profile: " + profile);
+    protected HandlerResult doAuthentication(final Credential credential) throws GeneralSecurityException, PreventedException {
+        final ClientCredential clientCredentials = (ClientCredential) credential;
+        logger.debug("clientCredentials  {}", clientCredentials);
+
+        final Credentials credentials = clientCredentials.getCredentials();
+        final String clientName = credentials.getClientName();
+        logger.debug("clientName:  {}", clientName);
+
+        // get client
+        final Client<Credentials, UserProfile> client = this.clients.findClient(clientName);
+        logger.debug("client: {}", client);
+
+        // web context
+        final ServletExternalContext servletExternalContext = (ServletExternalContext) ExternalContextHolder.getExternalContext();
+        final HttpServletRequest request = (HttpServletRequest) servletExternalContext.getNativeRequest();
+        final HttpServletResponse response = (HttpServletResponse) servletExternalContext.getNativeResponse();
+        final WebContext webContext = new J2EContext(request, response);
+
+        // get user profile
+        final UserProfile userProfile = client.getUserProfile(credentials, webContext);
+        logger.debug("userProfile : {}", userProfile);
```

Comment:
No I think it’s fine. But perhaps pac4j should not log this kind of thing, or obfuscate it. 

- Misagh

> On Oct 16, 2015, at 1:10 AM, LELEU Jérôme <notifications@github.com> wrote:
> 
> In cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/authentication/handler/support/ClientAuthenticationHandler.java <https://github.com/Jasig/cas/pull/1217#discussion_r42214364>:
> 
> > +        final String clientName = credentials.getClientName();
> > +        logger.debug("clientName:  {}", clientName);
> > +
> > +        // get client
> > +        final Client<Credentials, UserProfile> client = this.clients.findClient(clientName);
> > +        logger.debug("client: {}", client);
> > +
> > +        // web context
> > +        final ServletExternalContext servletExternalContext = (ServletExternalContext) ExternalContextHolder.getExternalContext();
> > +        final HttpServletRequest request = (HttpServletRequest) servletExternalContext.getNativeRequest();
> > +        final HttpServletResponse response = (HttpServletResponse) servletExternalContext.getNativeResponse();
> > +        final WebContext webContext = new J2EContext(request, response);
> > +
> > +        // get user profile
> > +        final UserProfile userProfile = client.getUserProfile(credentials, webContext);
> > +        logger.debug("userProfile : {}", userProfile);
> Credentials are the most sensitive and they are displayed appropriately (a few lines avode). We also consider some information like access token (OAuth) to be sensitive and they would appear here with this log.
> 
> So the risk is: if you turn on DEBUG on org.pac4j and read the log in the following hours (the access token lasts two hours for Facebook for example), you can take it and act on behalf of someone.
> 
> Do you want me to remove this log, change it to trace or keep it (as the risk is unlikely)?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/Jasig/cas/pull/1217/files#r42214364>.
>